### PR TITLE
Added POST /email-exists endpoint

### PR DESCRIPTION
**Pull request by kudrautsau-pavel**
- Added /email-exists URL in `urls.py` routing to `EmailExistsView`.
- Implemented `EmailExistsView` in `views.py` to check user email existence.
-  Implemented `EmailCheckSerializer`  in `serializers.py`

Aims to facilitate email existence checks within the core app.

### CODE REVIEW

File: apps/core/views.py [hunk 1]
```diff
@@ -412,3 +413,36 @@ def post(self, request, *args, **kwargs):
...
 
         return Response('Your password has been changed')
+
+
+class CheckEmailExistsView(APIView):
+    """
+    This view checks if a user with a certain email exists in the database.
+    """
+    queryset = User.objects.all()
+    permission_classes = (IsAuthenticated,)
```
Comment (sakalex) [inline apps/core/views.py:423]: this endpoint must be used before registration, so we can't check if the user is authorised

File: apps/core/views.py [hunk 2]
```diff
@@ -412,3 +413,36 @@ def post(self, request, *args, **kwargs):
...
+    def post(self, request, *args, **kwargs):
+        """
+        Checks the database for a user with the given email address.
+        Input: email
+        Result: A boolean indicating whether a user with the email exists
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        email = serializer.validated_data.get('email', '')  # Email is taken from serializer
+        # email = request.data.get('email', '')  - rewritten to get from serializer
```
Comment (sakalex) [inline apps/core/views.py:445]: unnecessary comment

**Comment by sakalex** [comment]
We also need to come up with a defence so that an attacker cannot endlessly search through random emails to find out which ones are registered with us.

File: apps/core/views.py (hunk 1 above)
Comment (kudrautsau-pavel) [reply apps/core/views.py]: fixed in a new commit

File: apps/core/views.py (hunk 2 above)
Comment (kudrautsau-pavel) [reply apps/core/views.py]: fixed in a new commit

**Comment by kudrautsau-pavel** [comment]
> We also need to come up with a defence so that an attacker cannot endlessly search through random emails to find out which ones are registered with us.

agreed with @jakkemomo  to skip this for now, as the problem will be resolved on infrastructure level

**Review by jakkemomo — APPROVED** [review]
Good job!

**Pull request merged**